### ISSUE TITLE
Allow the archiver block size to be user-configurable

### ISSUE THREAD

**Issue by rbeckman-nextgen** [nextgenhealthcare/connect#3427]
Right now we hard-code the archiver limit to 1000 and there's nothing the user can do to change it. If the server cannot load 1000 full Message objects (including all child connector messages and content, etc.) into memory at once, the user must increase the server's max heap size in order to be able to use the archiver. If increasing that size isn't possible, the user is not able to use the archiver at all.

[http://www.mirthcorp.com/community/forums/showthread.php?t=12201]

Imported Issue. Original Details:
Jira Issue Key: MIRTH-3531
Reporter: narupley
Created: 2014-11-19T08:09:53.000-0800

**Comment by rbeckman-nextgen** [comment]
The archiver block size is now configurable within the data pruner panel.

Imported Comment. Original Details:
Author: jaysenp
Created: 2014-12-30T16:35:14.000-0800

**Comment by rbeckman-nextgen** [comment]
Reverting the block size changes. Due to MIRTH-3581 there is currently no need for an archiver block size. Only one message will be cached at a time. In the future there are still ways we can reincorporate block size into the new algorithm, but there is no need to have the UI until that time.

Imported Comment. Original Details:
Author: wayneh
Created: 2015-01-21T14:54:44.000-0800

**Comment by rbeckman-nextgen** [comment]
Unreverting the block size changes. I've decided to modify the queries so that we actually retrieve multiple messages with a single query instead of just a single message. This will be an additional improvement as part of MIRTH-3581. Therefore increasing the archiver block size will now actually improve overall performance. However I still recommend leaving it set to its new default of 50.

Imported Comment. Original Details:
Author: wayneh
Created: 2015-01-26T13:08:53.000-0800

**Comment by rbeckman-nextgen** [comment]
I created some channels and populated them with test data for all databases. After taking a snapshot, I performed archiving with different block sizes (between 50-1000) and compared the tests. I was able to verify that increasing the size limit generally increased the messages archived per query as well as the overall speed of the process. I verified that the number of archived messages was expected in relation to how many messages actually contained content.

Imported Comment. Original Details:
Author: jaysenp
Created: 2015-01-28T14:40:45.000-0800
